src/besselj.jl

```diff
@@ -150,3 +150,203 @@ function besselj1(x::Float32)
+        elseif x <= nu
+            # this can produce inaccurate results when x is close to root of J0 or J1
+            #if isinteger(nu)
+            #    jnu = besselj1(x)
+            #    jnum1 = besselj0(x)
+            #    return besselj_up_recurrence(x, jnu, jnum1, 1.0, nu)[2]
+            #else
+                v = nu + ceil(nu*1.3)
+                jnu = besselj_debye(v, x)
+                jnup1 = besselj_debye(v+1, x)
+                return besselj_down_recurrence(x, jnu, jnup1, v, nu)[2]
+            #end
+        else
+            v = floor(nu - x/3)
+            v2 = nu - v
+            jnu = besselj_large_argument(v2, x)
+            jnum1 = besselj_large_argument(v2 -1, x)
+            return besselj_up_recurrence(x, jnu, jnum1, v2, nu)[2]
+        end
+    elseif 1.5*x < nu
+        return besselj_debye(nu, x)
+    elseif nu < 1000
+        v = nu + ceil(nu*1.3)
+        jnu = besselj_debye(v, x)
+        jnup1 = besselj_debye(v+1, x)
+        return besselj_down_recurrence(x, jnu, jnup1, v, nu)[2]
+    else
+        v = nu + 500
+        jnu = besselj_debye(v, x)
+        jnup1 = besselj_debye(v+1, x)
+        return besselj_down_recurrence(x, jnu, jnup1, v, nu)[2]
+    end
+end
+
+# for moderate size arguments of x and v this has relative errors ~9e-15
+# for large arguments relative errors ~1e-13
+function besselj_large_argument(v, x::T) where T
+    α, αp = _α_αp_asymptotic(v, x)
+    xn = fma(v, PIO2(T), PIO4(T))
+    b = SQ2OPI(T) / sqrt(αp * x)
+    return cos_sum(α, -xn)*b
+end
+
+# generally can only use for x < 4.0
+# this needs a better way to sum these as it produces large errors
+# only valid in non-oscillatory regime (v>1/2, 0<t<sqrt(v^2 - 0.25))
+# power series has premature underflow for large orders
+# gives wrong answers for x > 20.0 (might want to fix this)
+function besselj_small_arguments_orders(v, x::T) where T
+    MaxIter = 2000
+    out = zero(T)
+    a = (x/2)^v / gamma(v + one(T))
+    t2 = (x/2)^2
+    for i in 0:MaxIter
+        out += a
+        abs(a) < eps(T) * abs(out) && break
+        a *= -inv((v + i + one(T)) * (i + one(T))) * t2
+    end
+    return out
+end
+
+# this needs a better way to sum these as it produces large errors
+# use when v is large and x is small
+# need for bessely 
+function log_besselj_small_arguments_orders(v, x::T) where T
+    MaxIter = 1000
+    out = zero(T)
+    a = one(T)
+    x2 = (x/2)^2
+    for i in 0:MaxIter
+        out += a
+        a *= -x2 * inv((i + one(T)) * (v + i + one(T)))
+        (abs(a) < eps(T) * abs(out)) && break
+    end
+    logout = -loggamma(v + 1) + fma(v, log(x/2), log(out))
+    return exp(logout)
+end
+
+# valid when x < v (uniform asymptotic expansions)
+function besselj_debye(v, x)
+    T = eltype(x)
+    S = promote_type(T, Float64)
+    x = S(x)
+
+    vmx = fma(v,v, -x^2)
+    vdx = v/x
+    b = sqrt(vmx)
+
+    n = v * log(vdx + sqrt(vdx^2 - 1)) - b
+    coef = SQ1O2PI(S) * exp(-n) / sqrt(b)
```

Comment:
`coef` has some floating point errors that needs to be calculated more accurately in F64 for better accuracy